clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTemplateAtomicOperation.groovy

```diff
@@ -207,14 +213,25 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
       // Set the new instance template on the managed instance group.
       task.updateStatus BASE_PHASE, "Setting instance template $instanceTemplate.name on server group $serverGroupName..."
 
-      def instanceGroupManagersSetInstanceTemplateRequest =
+      if (isRegional) {
+        def regionInstanceGroupManagersSetTemplateRequest =
+          new RegionInstanceGroupManagersSetTemplateRequest(instanceTemplate: instanceTemplateUrl)
+        def setInstanceTemplateOperation = instanceGroupManagers.setInstanceTemplate(
+          project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest).execute()
+
+        // Block on setting the instance template on the managed instance group.
+        googleOperationPoller.waitForRegionalOperation(compute, project, region,
```

Comment:
Along the lines of Lars comment above, you could probably add a `waitForOperation` to the `googleOperationPoller` that allows the caller to specify the location, rather than having to replicate the call within the if block